src/components/gde/Middle.tsx

```diff
@@ -1,28 +1,35 @@
+"use client";
+
 import LeadCard from "../gdg/LeadCard";
-const experts = [
-  {
-    name: "John Doe",
-    city: "Los Angeles",
-    avatarSrc: "/images/test-avatar.png",
-  },
-  {
-    name: "Jane Smith",
-    city: "San Francisco",
-    avatarSrc: "/images/test-avatar.png",
-  },
-  {
-    name: "Alice Johnson",
-    city: "New York",
-    avatarSrc: "/images/test-avatar.png",
-  },
-  {
-    name: "Bob Wilson",
-    city: "Chicago",
-    avatarSrc: "/images/test-avatar.png",
-  },
-];
+import { useFirestore, useFirestoreCollectionData } from "reactfire";
+import { collection } from "firebase/firestore";
+import { useState, useEffect } from "react";
+
+type ExpertData = {
+  name: string;
+  city: string;
+  avatarSrc: string;
+};
 
 export default function Middle() {
+  const [expertData, setExpertData] = useState<ExpertData[]>([]);
+
+  const firestore = useFirestore();
+  const ref = collection(firestore, "gde-experts");
+  const { data } = useFirestoreCollectionData(ref, { idField: "id" });
+
+  useEffect(() => {
+    if (data) {
+      setExpertData(
+        data.map((doc: any) => ({
+          name: doc.name,
+          city: doc.city,
+          avatarSrc: doc.avatarSrc,
```

Comment:
does react fire provide generic types you can use instead of mapping it as any? for `collection<ExpertData>` example>